js/tests/unit/dropdown.spec.js

```diff
@@ -1783,4 +1783,125 @@ describe('Dropdown', () => {
+      '  <button class="btn dropdown-toggle" data-bs-toggle="dropdown">Dropdown</button>',
+      '  <div class="dropdown-menu">',
+      '    <form>',
+      '      <input class="form-control" type="text">',
+      '    </form>',
+      '  </div>',
+      '</div>'
+    ].join('')
+
+    const btnDropdown = fixtureEl.querySelector('[data-bs-toggle="dropdown"]')
+    const form = fixtureEl.querySelector('form')
+    const formInput = form.querySelector('input')
+    const formClickListener = jasmine.createSpy('formClickListener')
+
+    form.addEventListener('click', formClickListener)
+
+    btnDropdown.addEventListener('shown.bs.dropdown', () => {
+      formInput.click()
+      form.click()
+    })
+
+    btnDropdown.addEventListener('hidden.bs.dropdown', () => {
+      throw new Error('should not throw hidden.bs.dropdown event')
+    })
+
+    btnDropdown.click()
+
+    expect(formClickListener).toHaveBeenCalledTimes(2)
+  })
+
+  it('should open the dropdown when clicking `data-bs-toggle="dropdown"`', done => {
+    fixtureEl.innerHTML = [
+      '<div class="dropdown">',
+      '  <button class="btn dropdown-toggle" data-bs-toggle="dropdown">Dropdown</button>',
+      '  <div class="dropdown-menu">',
+      '    <a class="dropdown-item" href="#subMenu">Sub menu</a>',
+      '  </div>',
+      '</div>'
+    ].join('')
+
+    const triggerDropdown = fixtureEl.querySelector('[data-bs-toggle="dropdown"]')
+
+    spyOn(Dropdown, 'clearMenus').and.callThrough()
+    spyOn(triggerDropdown.classList, 'remove')
+
+    triggerDropdown.addEventListener('shown.bs.dropdown', () => {
+      setTimeout(() => {
+        expect(Dropdown.clearMenus).toHaveBeenCalledTimes(1)
+        expect(triggerDropdown.classList.remove).not.toHaveBeenCalled()
+        expect(triggerDropdown.classList.contains('show')).toEqual(true)
+        done()
+      }, 20)
+    })
+
+    triggerDropdown.addEventListener('hidden.bs.dropdown', () => {
+      throw new Error('should not throw hidden.bs.dropdown event')
+    })
+
+    triggerDropdown.click()
+  })
+
+  it('should open the dropdown when clicking the child element inside `data-bs-toggle="dropdown"`', done => {
+    fixtureEl.innerHTML = [
+      '<div class="container">',
+      '  <div class="dropdown">',
+      '    <button class="btn dropdown-toggle" data-bs-toggle="dropdown"><span id="childElement">Dropdown</span></button>',
+      '    <div class="dropdown-menu">',
+      '      <a class="dropdown-item" href="#subMenu">Sub menu</a>',
+      '    </div>',
+      '  </div>',
+      '</div>'
+    ].join('')
+
+    const triggerDropdown = fixtureEl.querySelector('[data-bs-toggle="dropdown"]')
+    const childElement = fixtureEl.querySelector('#childElement')
+
+    spyOn(Dropdown, 'clearMenus').and.callThrough()
+    spyOn(triggerDropdown.classList, 'remove')
+
+    triggerDropdown.addEventListener('shown.bs.dropdown', () => {
+      setTimeout(() => {
+        expect(Dropdown.clearMenus).toHaveBeenCalledTimes(1)
```

Comment:
You need to update this test IMO
`Dropdown.clearMenus` is called 2 times in real :slightly_smiling_face: